Hashlink.js

```diff
@@ -137,4 +176,29 @@ export class Hashlink {
     this.registeredTransforms[transform.algorithm] = transform;
   }
 
+  /**
+   * Finds a registered decoder for a given set of bytes or throws an Error.
+   *
+   * @param {Uint8Array} bytes - A stream of bytes to use when matching against
+   *   the registered decoders.
+   */
+  _findDecoder(bytes) {
+    const transforms = Object.values(this.registeredTransforms);
+    const decoder = transforms.reduce((decoder, transform) => {
+      let match = true;
+      let index = 0;
+      while(match && (index < transform.identifier.length)) {
+        match = transform.identifier[index] === bytes[index];
+        index++;
+      }
+      return (match) ? transform : decoder;
```

Comment:
Not really sure what's going on here ... but seems like `reduce` is the wrong method for it. It seems like a case for `.find` or `.every` or something? Perhaps some comments would help on what is supposed to happen.

Reply:
Not sure if this does the same thing/what is desired... but reading it is clearer to me:

For every transform, make sure every identifier matches what is in `bytes`, otherwise throw an error. Then return the last decoder (for some reason).

```js
const decoders = Object.values(this.registeredTransforms);
if(decoders.length === 0 ||
  !decoders.every(decoder => decoder.identifier.every((id, i) => id === bytes[i]))) {
  throw new Error('Could not determine decoder for: ' + bytes);
}
return decoders[decoders.length - 1];
```

Or perhaps you wanted to find any decoder with a list of identifiers that matched every byte?

```js
const decoders = Object.values(this.registeredTransforms);
const decoder = decoders.find(decoder => decoder.identifier.every((id, i) => id === bytes[i])));
if(!decoder) {
  throw new Error('Could not determine decoder for: ' + bytes);
}
return decoder;
```

Or is this supposed to match the length of the bytes and not just the length of the identifiers in the decoder, i.e., what if the bytes are longer than the identifiers... does that mean it shouldn't be a match?

Anyway, lots of questions here that are either because I'm tired or something else is up.

Reply:
It was the second one... implemented.